packages/next/server/app-render.tsx

```diff
@@ -28,8 +28,94 @@ import {
 import { FlushEffectsContext } from '../shared/lib/flush-effects'
 import { stripInternalQueries } from './internal-utils'
 import type { ComponentsType } from '../build/webpack/loaders/next-app-loader'
-import type { UnwrapPromise } from '../lib/coalesced-function'
 import { REDIRECT_ERROR_CODE } from '../client/components/redirect'
+import { NextCookies } from './web/spec-extension/cookies'
+
+const INTERNAL_HEADERS_INSTANCE = Symbol('internal for headers readonly')
+
+const readonlyHeadersError = new Error('ReadonlyHeaders cannot be modified')
```

Comment:
This would create the stack during module initialization which makes debugging these harder if you see them. It would be better to create a new Error object each time it's accessed.

Reply:
Done!